pkg/manifests/manifests.go

```diff
@@ -1424,12 +1425,41 @@ func (f *Factory) PrometheusK8s(grpcTLS *v1.Secret, telemetrySecret *v1.Secret)
 	for k := range p.Spec.RemoteWrite {
 		rw := &p.Spec.RemoteWrite[k]
 
+		var tmpRWProxyURL *string
 		if f.proxy.HTTPProxy() != "" {
-			rw.ProxyURL = ptr.To(f.proxy.HTTPProxy())
+			tmpRWProxyURL = ptr.To(f.proxy.HTTPProxy())
 		}
 		if f.proxy.HTTPSProxy() != "" {
-			rw.ProxyURL = ptr.To(f.proxy.HTTPSProxy())
+			tmpRWProxyURL = ptr.To(f.proxy.HTTPSProxy())
 		}
+
+		// Fixes: https://issues.redhat.com/browse/OCPBUGS-38289
+		// The code is condensed here to simplify backporting.
+		// This is a temporary workaround because the NoProxy field isn't exported on the Prometheus CRD in older versions.
+		// If the cluster proxy or remoteWrite config change, this code will re-run, ensuring no stale state.
+		// The httpproxy package used here is the same as used by Prometheus, which helps with consistency.
+
+		// proxy URL is not to be set/overwritten in here, nothing to do.
+		if tmpRWProxyURL == nil {
+			continue
+		}
+		// Suppose that CMO has set the proxy URL.
+		proxyConfig := &httpproxy.Config{
+			HTTPProxy:  *tmpRWProxyURL,
+			HTTPSProxy: *tmpRWProxyURL,
+			NoProxy:    f.proxy.NoProxy(),
+		}
+		proxyFunc := proxyConfig.ProxyFunc()
+		// Ignore errors here.
+		u, _ := url.Parse(rw.URL)
+		// Check if the endpoint URL will be using that proxy URL.
+		proxyURL, err := proxyFunc(u)
+		if err == nil && proxyURL == nil {
+			// Proxing will not be used, no need to set it.
+			continue
+		}
+		// The proxy will be used, set it.
+		rw.ProxyURL = tmpRWProxyURL
```

Comment:
```suggestion
		rw.ProxyURL = proxyURL.String()
```

Reply:
I think it's easier to understand the logic with `tmpRWProxyURL`
(if `noProxy` wouldn't change anything, we fallback to the old behaviour: CMO itself setting `rw.ProxyURL`)

Reply:
fair enough :)